### PR TITLE
[FIX] sale_mrp: fix picking creation issue for bom with kit type

### PR DESCRIPTION
**Pull request by gusi-odoo**
**Version:**

- saas-16.3

**Step to reproduce:**

- Create a subscription product with the type set to 'product'
- Create a BOM for that product with the BOM type set to 'kit'
- In the component, add any product with the type 'product'
- Create a sale order with the products that are created and generate an invoice
- Upon creating the next invoice for the subscription product, the picking was not being generated.

**Issue:**

Currently, if the product type is set to 'product' and recurring_invoice is
true, and the product has a bill of materials (BOM) with the type 'kit'
the picking was not created after the first invoice.

**Solution:**

If the product has a subscription, the quantity for BOM kits will not be updated.

task-3681597

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale_mrp/models/sale_order_line.py [hunk 1]
```diff
@@ -118,7 +118,7 @@ def _get_qty_procurement(self, previous_product_uom_qty=False):
         # We don't try to be too smart and keep a simple approach: we use the quantity of entire
         # kits that are currently in delivery
         bom = self.env['mrp.bom']._bom_find(self.product_id, bom_type='phantom')[self.product_id]
-        if bom:
+        if bom and not self.env.context.get('has_subscription', False):
```
[the remark this answers is not in the document]
Comment (gusi-odoo) [reply addons/sale_mrp/models/sale_order_line.py]: Hello @geflx ,

I've implemented the changes requested. Please have a look to this PR and the master branch.

Branch: master-subs-kit-issue-gusi
Odoo: https://github.com/odoo-dev/odoo/commit/248a68f6ac7289f673cc2bc2ef3e068e88a9365d
Enterprise: https://github.com/odoo-dev/enterprise/commit/64c2ab67870342db14cd74556d598ea77b06bbd

Thanks :innocent:

File: addons/sale_mrp/models/sale_order_line.py [hunk 2]
```diff
@@ -112,13 +112,17 @@ def _get_bom_component_qty(self, bom):
                 components[product] = {'qty': qty, 'uom': to_uom.id}
         return components
 
+    @api.model
```
[the remark this answers is not in the document]
Comment (gusi-odoo) [reply addons/sale_mrp/models/sale_order_line.py]: Hello @geflx ,
Done the changes 
Thanks for the suggestion :innocent:

File: addons/sale_mrp/models/sale_order_line.py [hunk 3]
```diff
@@ -112,6 +112,19 @@ def _get_bom_component_qty(self, bom):
                 components[product] = {'qty': qty, 'uom': to_uom.id}
         return components
 
+    def _get_incoming_outgoing_moves_filter(self):
+        """ Method to be override: will get incoming moves and outgoing moves.
+
+        :return: Dictionary with incoming moves and outgoing moves
+        :rtype: dict
+        """
+        self.ensure_one()
```
Comment (Whenrow) [inline addons/sale_mrp/models/sale_order_line.py:121]: remove `ensure_one()` and use `@api.model` decorator instead

**Review by Whenrow — CHANGES_REQUESTED** [review]

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**